Code/GraphMol/Fingerprints/MHFP.h

```diff
@@ -0,0 +1,302 @@
+
+  //! \overload
+  std::vector<std::vector<uint32_t>>
+  Encode(std::vector<std::string>& smiles,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1);
+
+
+  /*!
+    \brief Creates a binary fingerprint based on circular sub-SMILES.
+   
+    Creates a binary fingerprint similar to ECFP. However, instead of using
+    a Morgan-style hashing, circular n-grams (sub-SMILES) are created, hashed
+    directly and folded.
+   
+    \param radius the maximum radius of the substructure that is generated at
+          each atom. Default: <tt>3</tt>.
+    \param rings whether the rings (SSSR) are extrected from the molecule and
+           added to the shingling. Given the molecule 
+           <tt>"C1CCCCCC1C(=O)C"</tt>, "<tt>C1CCCCCC1"</tt> would be added
+           to the shingling. Default: <tt>true</tt>.
+    \param isomeric whether the SMILES added to the shingling are isomeric.
+            Default: <tt>false</tt>.
+    \param kekulize whether the SMILES added to the shingling are kekulized.
+            Default: <tt>true</tt>.
+    \param min_radius the minimum radius that is used to extract n-grams.
+            Default: <tt>1</tt>.
+    \param length the length into which the fingerprint is folded.
+            Default: <tt>2048</tt>.
+
+    \returns the SECFP fingerprint.
+   */
+  ExplicitBitVect
+  EncodeSECFP(ROMol& mol,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  //! \overload
+  std::vector<ExplicitBitVect>
+  EncodeSECFP(std::vector<ROMol>& mols,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  //! \overload
+  ExplicitBitVect
+  EncodeSECFP(std::string& smiles,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  //! \overload
+  std::vector<ExplicitBitVect>
+  EncodeSECFP(std::vector<std::string>& smiles,
+       const unsigned char& radius = 3,
+       const bool& rings = true,
+       const bool& isomeric = false,
+       const bool& kekulize = true,
+       const unsigned char& min_radius = 1,
+       const size_t& length = 2048);
+
+  /*!
+    \brief Calculates the Jaccard / Tanimoto distance between two MHFP fingerprints.
+   
+    \param a an MHFP fingerprint vector.
+    \param b an MHFP fingerprint vector.
+
+    \returns the Jaccard / Tanimoto distance between the two fingerprints.
+   */
+  static float
```

Comment:
does this need to return `float`? Couldn't it be `double`? On modern CPUs, it is often faster to work with doubles.

Reply:
How about including the definition of the method here instead of in the cpp file?